pkg/k8s/operator/status/agent_status_manager.go

```diff
@@ -262,6 +278,10 @@ func (a *agentStatusManager) getAllK8sSensorsAvailableCondition(ctx context.Cont
 	return result.OfSuccess(condition)
 }
 
+func objectKeyHasName(key client.ObjectKey) bool {
+	return key.Name != ""
```

Comment:
as mentioned, since you're effectively checking the presence of the inverse of this, just check the inverse of this i.e: `key.Name == ""` and rename the function to reflect so